src/shapes/Polygon.spec.ts

```diff
@@ -0,0 +1,568 @@
+} from '../../fabric';
+import { Polygon } from './Polygon';
+
+function getPoints() {
+  return [
+    { x: 10, y: 12 },
+    { x: 20, y: 22 },
+  ];
+}
+
+const REFERENCE_OBJECT = {
+  version: version,
+  type: 'Polygon',
+  originX: 'left',
+  originY: 'top',
+  left: 9.5,
+  top: 11.5,
+  width: 10,
+  height: 10,
+  fill: 'rgb(0,0,0)',
+  stroke: null,
+  strokeWidth: 1,
+  strokeDashArray: null,
+  strokeLineCap: 'butt',
+  strokeDashOffset: 0,
+  strokeLineJoin: 'miter',
+  strokeMiterLimit: 4,
+  scaleX: 1,
+  scaleY: 1,
+  angle: 0,
+  flipX: false,
+  flipY: false,
+  opacity: 1,
+  points: getPoints(),
+  shadow: null,
+  visible: true,
+  backgroundColor: '',
+  fillRule: 'nonzero',
+  paintFirst: 'fill',
+  globalCompositeOperation: 'source-over',
+  skewX: 0,
+  skewY: 0,
+  strokeUniform: false,
+} as const;
+
+const REFERENCE_EMPTY_OBJECT = {
+  points: [],
+  width: 0,
+  height: 0,
+  top: 0,
+  left: 0,
+};
+
+describe('Polygon', () => {
+  it('constructor', () => {
+    expect(Polygon, 'Polygon class should exist').toBeTruthy();
+
+    const polygon = new Polygon(getPoints());
+
+    expect(polygon, 'should be instance of Polygon').toBeInstanceOf(Polygon);
+    expect(polygon, 'should be instance of Polyline').toBeInstanceOf(Polyline);
+    expect(polygon, 'should be instance of FabricObject').toBeInstanceOf(
+      FabricObject,
+    );
+
+    expect(polygon.constructor, 'type should be Polygon').toHaveProperty(
+      'type',
+      'Polygon',
+    );
+    expect(polygon.get('points'), 'points should match input').toEqual([
+      { x: 10, y: 12 },
+      { x: 20, y: 22 },
+    ]);
+  });
+
+  it('constructor, with strokeWidth top-left and origins top-left', () => {
+    const polygon = new Polygon(getPoints(), {
+      strokeWidth: 2,
+      originX: 'left',
+      originY: 'top',
+    });
+
+    expect(polygon.left, 'left should be 9').toBe(9);
+    expect(polygon.top, 'top should be 11').toBe(11);
+  });
+
+  it('constructor, with strokeWidth top-left and origins center-center', () => {
+    const polygon = new Polygon(getPoints(), {
+      strokeWidth: 2,
+      originX: 'center',
+      originY: 'center',
+    });
+
+    expect(polygon.left, 'left should be 15').toBe(15);
+    expect(polygon.top, 'top should be 17').toBe(17);
+  });
+
+  it('constructor, with strokeWidth top-left and origins bottom-right', () => {
+    const polygon = new Polygon(getPoints(), {
+      strokeWidth: 2,
+      originX: 'right',
+      originY: 'bottom',
+    });
+
+    expect(polygon.left, 'left should be 21').toBe(21);
+    expect(polygon.top, 'top should be 23').toBe(23);
+  });
+
+  it('polygon with exactBoundingBox false', () => {
+    const polygon = new Polygon(
+      [
+        { x: 10, y: 10 },
+        { x: 20, y: 10 },
+        { x: 20, y: 100 },
+      ],
+      {
+        // @ts-expect-error -- TODO: are types wrong for Polygon? seems like it doesn't accept exactBoundingBox property
+        exactBoundingBox: false,
```

Comment:
@asturur seems like typescript doesn't accept exactBoundingBox as a param here
this will also reflect for users of the fabric
either this prop is redundant or types should be improved to include this prop